fili-core/src/main/java/com/yahoo/bard/webservice/table/resolver/PhysicalDataSourceConstraint.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.resolver;
+
+import com.yahoo.bard.webservice.table.PhysicalTable;
+import com.yahoo.bard.webservice.table.PhysicalTableSchema;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Data source constraint containing physical name of the columns.
+ */
+public class PhysicalDataSourceConstraint extends DataSourceConstraint {
+
+    private final Set<String> allColumnPhysicalNames;
+
+    /**
+     * Constructor.
+     *
+     * @param dataSourceConstraint  Data soruce contraint containing all the column names as logical names
+     * @param physicalTableSchema  A map from logical column name to physical column names
+     */
+    public PhysicalDataSourceConstraint(DataSourceConstraint dataSourceConstraint, PhysicalTableSchema physicalTableSchema) {
+        super(dataSourceConstraint);
+        this.allColumnPhysicalNames = dataSourceConstraint.getAllColumnNames().stream()
+                .map(physicalTableSchema::getPhysicalColumnName)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Getter for the all column names as physical names.
+     *
+     * @return the physical name of all the columns
+     */
+    public Set<String> getAllColumnPhysicalNames() {
```

Comment:
Should we rename this `physicalColumnNames`? `columnPhysicalNames` feels a bit backwards, for some reason.